compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -918,9 +918,9 @@ abstract class GenJSCode extends plugins.PluginComponent
        */
       implicit val pos = tree.pos
       tree match {
-        case js.Block(stats :+ expr)  => js.Block(stats :+ exprToStat(expr))
-        case _:js.Literal | js.This() => js.Skip()
-        case _                        => tree
+        case js.Block(stats :+ expr)   => js.Block(stats :+ exprToStat(expr))
+        case _: js.Literal | js.This() => js.Skip()
```

Comment:
Actually, this is an intentional departure from the general rule of "space after :". When used in a type-match with alternatives (with `|`), we don't put a space after the colon, so that it is clearer that the colon has higher precedence than the `|`.